tasks/tasks.py

```diff
@@ -63,3 +63,103 @@ def run(self):
         scim_c.import_all_users_and_groups()
         end = timer()
         logging.info("Complete Users and Groups Import Time: " + str(timedelta(seconds=end - start)))
+
+class WorkspaceItemLogTask(AbstractTask):
```

Comment:
It would be great if we can have a separate PR for adding new tasks.

Both checkpointing and new tasks are very important, mixing them together will slow down both and we might overlook some issues.

Reply:
One of main reason I added this WorkspaceItemTask as a part of this PR is to create visibility of how the checkpoint service can be used for pipeline task level checkpointing. (Since UserExport/ImportTask didn't require checkpointing).
TBH, the logic of adding new task is fairly straightforward, we are merely copying the logic from export_db.py and import_db.py. There is no new logic which is added at this point. 

what do you think ?

Reply:
I agree it makes sense to keep WorkspaceClient changes in this PR to demonstrate how checkpointing framework is used. Also I'm ok to keep checkpointing related changes in pipeline framework.

But adding WorkspaceItemTask is orthogonal to both intra-task checkpointing (changes in WorkspaceClient) and inter-task checkpointing (changes in pipeline framework). So I think it's better to exclude them from this PR.

Reply:
moved them out of PR